backend/app/routes/admin/putAdmin.js

```diff
@@ -0,0 +1,28 @@
+const to = require('await-to-js').default;
+const Admin = require('../../models/Admin');
+const { ErrorHandler } = require('../../../helpers/error');
+const constants = require('../../../constants');
+
+module.exports = async (req, res, next) => {
+  const { firstName, lastName, contact, username } = req.body;
+  const userRecord = await Admin.findOne({
+    email: res.locals.decode.email,
+  });
+
+  try {
+    await to(Admin.findOneAndUpdate({ email: userRecord.email }, { $set: { firstName, lastName, contact, username } }));
+    const response = await Admin.findOne({
+      email: res.locals.decode.email,
+    });
+    res.status(201).send(response);
+    return next();
+  } catch (err) {
```

Comment:
There is a logical fault here, What is I try to update my usename with something which is already assigned to someone. How will the frontend deduce that?